src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -345,5 +399,49 @@ public static void MapToolTip(IViewHandler handler, IView view)
 				handler.ToPlatform().UpdateToolTip(tooltipContainer.ToolTip);
 #endif
 		}
+
+		void OnChangedContainerCollectionChanged(object? sender, EventArgs e) =>
+			UpdateValue(nameof(IDynamicContainerViewHandler.NeedsContainer));
+
+		class CollectionWithEvents : ICollection<string>
+		{
+			HashSet<string> properties = new();
+
+			public event EventHandler? Changed;
+
+			public int Count => properties.Count;
+
+			public bool IsReadOnly => false;
+
+			public void Add(string item)
+			{
+				if (properties.Add(item))
+					Changed?.Invoke(this, EventArgs.Empty);
+			}
+
+			public void Clear()
+			{
+				properties.Clear();
```

Comment:
A double clear will fire the event twice, the first of a clear and the second for a clear of an empty collection, is that intended?

Reply:
Nope and fixed!